notebooks/05-operations/02-druid-logs-asynch.ipynb

```diff
@@ -0,0 +1,381 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "151ede96-348f-423c-adc7-0e23c6910b9e",
+   "metadata": {},
+   "source": [
+    "## Initialization\n",
+    "\n",
+    "To use this notebook, you will need to run Druid locally.\n",
+    "\n",
+    "You will also make extensive use of the terminal, which we suggest you place alongside this notebook or on another screen.\n",
+    "\n",
+    "### Install tools\n",
+    "\n",
+    "Open a local terminal window.\n",
+    "\n",
+    "If you have not already install `wget` or `multitail`, run the following commands to install these tools using `brew`.\n",
+    "\n",
+    "```bash\n",
+    "brew install multitail && brew install wget\n",
+    "```\n",
+    "\n",
+    "Run the following command to pull the default configuration for `multitail` to your home folder. Do not run this command if you are already running `multitail` as it will overwrite your own configuration.\n",
+    "\n",
+    "```bash\n",
+    "curl https://raw.githubusercontent.com/halturin/multitail/master/multitail.conf > ~/.multitailrc\n",
+    "```\n",
+    "\n",
+    "### Install Apache Druid\n",
+    "\n",
+    "Run the following to create a dedicated folder for learn-druid in your home directory:\n",
+    "\n",
+    "```bash\n",
+    "cd ~ && mkdir learn-druid-local\n",
+    "cd learn-druid-local\n",
+    "```\n",
+    "\n",
+    "Now pull a compatible version of Druid.\n",
+    "\n",
+    "> If you do not have wget on your Mac, you can install it with brew.\n",
+    "\n",
+    "```bash\n",
+    "wget https://dlcdn.apache.org/druid/28.0.1/apache-druid-28.0.1-bin.tar.gz && tar -xzf apache-druid-28.0.1-bin.tar.gz && mv apache-druid-28.0.1 apache-druid\n",
+    "cd apache-druid\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65384957-3c3f-43fb-b296-7e5f7fddbcc5",
+   "metadata": {},
+   "source": [
+    "## Process logs\n",
+    "\n",
+    "Understand [SQL-based ingestion](https://druid.apache.org/docs/latest/multi-stage-query/concepts.html) planning and execution by monitoring logs emitted from the [core processes for ingestion](https://druid.apache.org/docs/latest/ingestion/):\n",
+    "\n",
+    "* Router\n",
+    "* Overlord\n",
+    "* MiddleManager\n",
+    "* Historical\n",
+    "\n",
+    "Start Druid as a background task with the following command:\n",
+    "\n",
+    "```bash\n",
+    "nohup ~/learn-druid-local/apache-druid/bin/start-druid && disown > log.out 2> log.err < /dev/null && disown\n",
+    "```\n",
+    "\n",
+    "In the terminal window, switch to the default log location:\n",
+    "\n",
+    "```bash\n",
+    "cd log\n",
+    "```\n",
+    "\n",
+    "Run the following command to start monitoring these logs. For the purpose of this exercise, the command includes the `-ev` flag to exclude some log messages from the Coordinator.\n",
+    "\n",
+    "```bash\n",
+    "multitail -CS log4jnew -du -P a -s 2 -sn 1,2 \\\n",
```

Comment:
Is there a config to stop these messages, they are noisy for the example: 
```
Debug: 12 hidden stats. Set 'debugDimensions' to see these.  
TOTAL: 12 stats for 9 dimension keys       

Debug: 12 hidden stats. Set 'debugDimensions' to see these.  
TOTAL: 12 stats for 9 dimension keys       

Debug: 12 hidden stats. Set 'debugDimensions' to see these.  
TOTAL: 12 stats for 9 dimension keys       

...

```

Reply:
These are weird extra lines being emitted by the metrics emitter - it looks like multitail doesn't like there being a new line, so it's keeping them in. <sigh> I've therefore removed the filter because it just looked weird.